docs/README.md

```diff
@@ -44,20 +44,81 @@ $ wibeth_tpg_algorithms_emulator -f swtest_run000035_0000_dataflow0_datawriter_0
+They have several parameters to customize the plots, in particular in some cases the range is handled through the quantile of the distribution. 
+Default value is 1, but to get rid of outliers it can be set to 0.9, for example.
+
+* `group_maker.py` is a library to group TPs basing on channel and/or time. 
+Each group should then be a track, if the conditions are set to be strict enough (deafult is channel limit 1 and time limit 3 ticks).
+
+* `image_creator.py` contains functions to create images from TPs, that will be 2D histograms (channel vs time). 
+Also in this case, many parameters can be set to customize the images, for example to choose if to fix the size or if it will depend on the group case by case.
+
+
+### Python scripts
+In the `scripts` directory, there are some example scripts that make useof the python libraries.
+To avoid unwanted files in the repo, I added `*.png` to the `.gitignore` file.
+Taking these as inspiration, you can create your own scripts to perform the analyses you need.
+They have a verbose option (`-v`) that prints out the TPs that are being processed.
+
+#### `plot_tp_properties.py` 
+
+Script to plot all the properties of the TPs: start time, time over threshold, channel, detid, peak, peak time, integral. 
+It can accept one or more files are input, both in hdf5 or text format. 
+The flag `--superimpose` can be used to plot all the input files on the same plot. 
+Default output path is `./`, output files will be named `input_file_timeStart.png` and so on.
+You can see all options with 
+```sh
+python plot_tp_properties.py --help
+```
+
+Here an example of usage. `--all` is used to plot all the properties, but you can also choose to plot only one or some of them.
+```sh
+python plot_tp_properties.py -f INPUT_TPSTREAM.hdf5 INPUT_TPS.txt -n 1000 -e my/output/folder/ --superimpose --all
+```
+
+
+#### `create_images.py` 
+Script to create images from TP groups, ie tracks or clusters of tracks. 
+It accepts one input file (hdf5 or text) and will produce images with increasing numbering in the output directory (default is `./`).
+Default detector is "APA", but it can be changed with the flag `--detector` to "CRP" or "50L".
+The images are by default `png`, named `<view>_track<number>.png`, e. g. `u_track12.png`.
+The groups can be saved to a text file to take a look at them in a readable format, with the flag `--save-groups`.
+
+You can see full usage with `--help`, here an example:
+```sh
+python create_images.py -i INPUT_TPSTREAM.hdf5 -n 1000 -o my/output/folder/ --ticks-limit 5 --channels-limit 2 --min-tps 3 
+```
+
+
+#### Setup DAQ environment on lxplus or NP04 machines (e.g. `np04-srv-019`)
+To use the tools and scripts in this repository, the DUNE-DAQ software environment must be setup. The following commands are valid for lxplus machines and NP04 machines (e.g. `np04-srv-019`). 
+```sh
+source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
+setup_dbt latest
+dbt-setup-release fddaq-v4.1.1
```

Comment:
Since we already have it, let's use the last DUNE-DAQ release as an example: dbt-setup-release fddaq-v4.2.0

Reply:
Done in 2eebd2c